src/automations/components/AutomationActionSendNotificationInput.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <p-content class="automation-action-send-notification">
+    <p-label label="Block" :state="blockDocumentState" :message="blockDocumentError">
+      <BlockCapabilityBlockDocumentInput v-model="blockDocumentId" :state="blockDocumentState" capability="notify" required use-modal />
+    </p-label>
+
+    <template v-if="isAnonymousBlock">
+      <p-message info>
+        This notification action is using an anonymous block. This is most likely because it was migrated from an existing notification.
+      </p-message>
+    </template>
+
+    <p-label label="Subject" :state="subjectState" :message="subjectError">
+      <p-code-input v-model="subject" :state="subjectState" lang="jinja" />
+    </p-label>
+
+    <p-label label="Body" :state="bodyState" :message="bodyError">
+      <p-code-input v-model="body" :state="bodyState" rows="5" lang="jinja" />
+    </p-label>
+
+    <p-message info>
+      Event fields and any of the following objects present in the triggering event can be used in notification templates: flow, deployment, flow run, work pool, work queue, metric.
+    </p-message>
+  </p-content>
+</template>
+
+<script lang="ts" setup>
+  import { useValidation, ValidationRule } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import { AutomationActionSendNotification } from '@/automations/types/actions'
+  import BlockCapabilityBlockDocumentInput from '@/components/BlockCapabilityBlockDocumentInput.vue'
+  import { useBlockDocument, useWorkspaceApi } from '@/compositions'
+  import { timeout } from '@/utilities/time'
+  import { isRequired } from '@/utilities/validation'
+
+  const props = defineProps<{
+    action: Partial<AutomationActionSendNotification>,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'update:action', value: Partial<AutomationActionSendNotification>): void,
+  }>()
+
+  const api = useWorkspaceApi()
+
+  const isValidSubject: ValidationRule<string> = async (value, label, options) => {
+    return await isValidTemplate(value, label, options)
+  }
+
+  const isValidBody: ValidationRule<string> = async (value, label, options) => {
+    return await isValidTemplate(value, label, options)
+  }
+
+  const isValidTemplate: ValidationRule<string> = async (value, label, { signal, source, previousValue }) => {
+    if (value === previousValue) {
+      return
+    }
+
+    if (source === 'validator') {
+      await timeout(3000, signal)
+    }
+
+    if (signal.aborted) {
+      return
+    }
+
+    const valid = await api.automations.validateTemplate(value)
```

Comment:
recognizing this doesn't exist in the public API yet, can you still add it to the client-side API class?

Reply:
Yes added in https://github.com/PrefectHQ/prefect-ui-library/pull/2281/commits/0bfa2f00ac731b79aed40c56b5b99c1d1b03036c

Reply:
You were super fast! ❤️